subsys/net/l2/openthread/openthread_utils.c

```diff
@@ -19,6 +19,56 @@
 
 #include "openthread_utils.h"
 
+int hex2bin(const char *aHex, uint8_t *aBin, uint16_t aBinLength)
```

Comment:
We have very similar function already in include/net/net_ip.h called net_bytes_from_str(). Could you make it work with plain hex strings too (it now parses strings in "aa:bb:cc:dd" format) so that we would not need to introduce similar kind of function.